### PR TITLE
Feat/integrate location web

### PR DESCRIPTION
**Pull request by javidaf**
Extracted GPS data from mobile and published the data to IoT topic "location/topic",
Subscribe to IoT  Topic "location/topic" sendt from mobile and integrated the location data into a map system.

Closes: #73

### CODE REVIEW

File: norbit-webapp/components/Map/Map.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+// Deferred import for Leaflet
+let L: typeof import('leaflet');
+if (typeof window !== 'undefined') {
+  L = require('leaflet');
+}
+
+
+const LeafletMap: React.FC<{ amplifyInstance: TamplifyInstance | null }> = ({ amplifyInstance }) => {
```
Comment (OliviaWork) [inline norbit-webapp/components/Map/Map.tsx:16]: In line with our other component naming (and repo conventions), this should be named MapComponent, and the file and folder should also be called MapComponent.

File: norbit-webapp/components/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -51,6 +52,7 @@ const Dashboard: React.FC = () => {
         {sensorConfig.temperature && <TemperatureComponent amplifyInstance={mockAmplify ? null : Amplify} />}
         {sensorConfig.light && <LightIntensityComponent amplifyInstance={mockAmplify ? null : Amplify} />}
         {sensorConfig.sound && <SoundLevelComponent amplifyInstance={mockAmplify ? null : Amplify} />}
+        <Map amplifyInstance={mockAmplify ? null : Amplify} />
```
Comment (OliviaWork) [inline norbit-webapp/components/dashboard/Dashboard.tsx:55]: SensorConfig needs to be extend with `location`, and this line needs to be
```suggestion
        {sensorConfig.location && <Map amplifyInstance={mockAmplify ? null : Amplify} />}
```

File: norbit-webapp/pages/index.tsx [hunk 3]
```diff
@@ -1,6 +1,8 @@
-import Head from 'next/head'
-import styles from '@styles/Home.module.css'
-import Dashboard from 'components/dashboard/Dashboard'
+import Head from 'next/head';
+import styles from '../styles/Home.module.css';
+import Dashboard from 'components/dashboard/Dashboard';
+
+
```
Comment (OliviaWork) [inline norbit-webapp/pages/index.tsx:5]:
```suggestion
```

File: norbit-webapp/pages/index.tsx [hunk 4]
```diff
@@ -15,6 +17,9 @@ export default function Home() {
       <main className={styles.main}>
         <Dashboard />
 
+
+
+
```
Comment (OliviaWork) [inline norbit-webapp/pages/index.tsx:22]:
```suggestion
```

File: norbit-webapp/components/Map/Map.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  useEffect(() => {
+    if (!mapRef.current) return;
+    if (mapInstance.current) return;  // Return if map is already initialized
+
+    // Create the map instance
+    mapInstance.current = L.map(mapRef.current).setView([mapData.latitude,mapData.longitude], 13);
+
+
+    // Add a marker
+    L.marker([mapData.latitude,mapData.longitude], {icon: greenIcon}).addTo(mapInstance.current);
```
Comment (OliviaWork) [inline norbit-webapp/components/Map/Map.tsx:50]: If `L.icon` needs a null check for `L`, why doesn't this?

Comment (OliviaWork) [reply norbit-webapp/components/Map/Map.tsx:50]: and a null-check for greenIcon

File: norbit-webapp/components/Map/Map.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+
+const LeafletMap: React.FC<{ amplifyInstance: TamplifyInstance | null }> = ({ amplifyInstance }) => {
+
+  const mapRef = useRef(null);
+  const mapInstance = useRef<L.Map | null>(null); 
+  const [payload, setPayload] = useState<any>(null); 
+  const [mapData, setMapData] = useState<MapData>({ latitude: 63, longitude: 10 });
+  console.log(mapData);
```
Comment (OliviaWork) [inline norbit-webapp/components/Map/Map.tsx:22]:
```suggestion
```

**Review by OliviaWork — CHANGES_REQUESTED** [review]

File: norbit-webapp/components/Map/Map.tsx (hunk 1 above)
Comment (javidaf) [reply norbit-webapp/components/Map/Map.tsx:16]: makes sense, well done!

File: norbit-webapp/components/Map/Map.tsx (hunk 5 above)
Comment (javidaf) [reply norbit-webapp/components/Map/Map.tsx:50]: makes sense, well done!

**Pull request merged**